### PR TITLE
Migrate mentee API calls to mentee-services

### LINKED ISSUE #191: Move mentee APIs to the mentee-services file in ManageMentees scene
**Is your feature request related to a problem? Please describe.**
- According to the current implementation we are using Axios call inside our JSX files. 
- Move the APIs in the ManageMentees scene to a single file.

**Describe the solution you'd like**
- Create the `mentee-services.ts` file 
- Move all the mentee related APIs to the created file

https://sef.discourse.group/t/separating-the-api-calls-from-the-jsx-files/310

https://github1s.com/sef-global/scholarx-frontend/blob/HEAD/src/scenes/Dashboard/scenes/ManageMentees/index.tsx#L58

### PR DESCRIPTION
**Pull request by NethmiRodrigo**
## Purpose
The purpose of this PR is to fix #191 

## Goals
To migrate all mentee related API calls to a separate utility file

## Approach
Removed axios calls related to mentees from all components and moved them to a separate utility file called mentee-services

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Test environment
Windows 10
Google Chrome

### CODE REVIEW

File: src/scenes/Home/scenes/RequestMentors/scenes/MenteeApplication/index.tsx [hunk 1]
```diff
@@ -32,53 +37,26 @@ function MenteeApplication() {
...
-        setIsLoading(false);
-        notification.error({
-          message: error.toString(),
-          description: 'Something went wrong when fetching the mentee',
-        });
-      });
-  }, []);
+  const getApplicationDetails = async () => {
+    const mentee: Mentee = await getMenteeApplication(mentorId);
+    console.log(mentee);
```
Comment (anjula-sack) [inline src/scenes/Home/scenes/RequestMentors/scenes/MenteeApplication/index.tsx]: Let's remove the console log

File: src/scenes/Home/scenes/RequestMentors/scenes/MenteeApplication/index.tsx [hunk 2]
```diff
@@ -88,34 +66,25 @@ function MenteeApplication() {
...
+      setIsFormVisible(false);
+      if (statusCode == 201) {
+        setIsApplied(true);
+      }
+      notification.success({
+        message: 'Success!',
+        description: 'Successfully applied!',
       });
+    } else {
+      console.log(response);
```
Comment (anjula-sack) [inline src/scenes/Home/scenes/RequestMentors/scenes/MenteeApplication/index.tsx]: Here as well

**Review by anjula-sack — CHANGES_REQUESTED** [review]
Awesome job! There are some small changes

File: src/scenes/Home/scenes/RequestMentors/scenes/MenteeApplication/index.tsx (hunk 1 above)
Comment (NethmiRodrigo) [reply src/scenes/Home/scenes/RequestMentors/scenes/MenteeApplication/index.tsx]: I forgot to remove it sorry😅.

**Review by anjula-sack — CHANGES_REQUESTED** [review]
Can you resolve the conflicts? @NethmiRodrigo

**Comment by cmdrGuyson** [comment]
@NethmiRodrigo I think we should close this PR. There have been a lot of changes (specially with my latest PR). Lets do this refactor while wiring the new UIs.

@anjula-sack I think we should make sure all network call logic is inside a service file when approving PRs with business logic

**Comment by jayasanka-sack** [comment]
> @NethmiRodrigo I think we should close this PR. There have been a lot of changes (specially with my latest PR). Lets do this refactor while wiring the new UIs.
> 
> @anjula-sack I think we should make sure all network call logic is inside a service file when approving PRs with business logic

Agree with @cmdrGuyson's suggestion.

**Pull request closed without merging**